### ISSUE TITLE
Something fishy with number of events logged

### ISSUE THREAD

**Issue by miklosduma** [sparkl/cli#22]
1. Import the sample mix below
2. Run `sparkl listen -c | sparkl elastic`
3. Run `sparkl call Scratch/hello_world/Mix/Start`
4. Ctrl-x out of sparkl listen
--> Kibana does not show any events logged
5. Go through the same steps but run the mix twice before hitting ctrl-x
--> 6 events generated
6. Go through the same steps but run the mix three times
--> 14 events generated, also, number of transaction start and end events does not add up.

What you should be getting per transaction:
* 1 txn_start event
* 1 txn_end event
* 4 data events (Start, SendMessage, Ok, Ok)

```xml
<folder name="hello_world">
  <service name="Expr" provision="expr"/>
  <service name="Sequencer" provision="sequencer"/>
  <field name="START"/>
  <field name="message" type="string"/>
  <mix name="Mix">
    <solicit name="Start" service="Sequencer" fields="START">
      <response name="Ok" fields="message"/>
    </solicit>
    <request name="SendMessage" service="Expr" fields="START">
      <prop name="expr.bind.out" Message="message"/>
      <prop name="expr.src" content-type="text/x-erlang"><![CDATA[
Message = "Hello, World!",
  "Ok".
      ]]></prop>
      <reply name="Ok" fields="message"/>
    </request>
  </mix>
</folder>
```

**Comment by jacobythwaites** [comment]
@miklosduma By ctl-x, do you mean ctl-c?

**Comment by miklosduma** [comment]
Yep sure, sorry :)

**Comment by jacobythwaites** [comment]
Fixed by #25
